### PR TITLE
Add exclusive lock to save/load status

### LINKED ISSUE #1113: update_saved_status should acquire exclusive lock
We should acquire a write-lock.

- https://github.com/jubatus/jubatus/blob/0.9.0/jubatus/server/framework/server_base.cpp#L184
- https://github.com/jubatus/jubatus/blob/0.9.0/jubatus/server/framework/server_base.cpp#L221-L224

### PR DESCRIPTION
**Pull request by rimms**
Fixes #1113

### CODE REVIEW

File: jubatus/server/framework/server_base.hpp [hunk 1]
```diff
@@ -76,18 +76,22 @@ class server_base {
   }
 
   uint64_t last_saved_sec() const {
+    jubatus::util::concurrent::scoped_rlock lock(status_mutex_);
     return last_saved_.sec;
   }
 
   const std::string& last_saved_path() const {
```
Comment (kmaehashi) [inline jubatus/server/framework/server_base.hpp]: Is it safe to return pointer of the protected resource? (same for `last_loaded_path`)

Comment (rimms) [inline jubatus/server/framework/server_base.hpp]: Fixed.

**Comment by rimms** [comment]
@kmaehashi Thanks for review, fixed.

**Comment by kmaehashi** [comment]
:+1:

**Pull request merged**